include/alpaka/core/Sycl.hpp

```diff
@@ -23,6 +23,36 @@
 
 #    include <CL/sycl.hpp>
 
+// if SYCL is enabled with the AMD backend the printf will be killed because of missing compiler support
+#    ifdef __AMDGCN__
+#        include <cstdio> // the define breaks <cstdio> if it is included afterwards
```

Comment:
Please move the `#include` outside the preprocessor guard to not break amalgamation of alpaka headers.

Reply:
do you mean:
```cpp
#        include <cstdio> 
#    ifdef __AMDGCN__
#        define printf(...)
#    else
```
?
The `#include <cstdio>` is needed only for AMD that does not have its own `printf`

Reply:
ok, I'll move it out then